gateway-js/src/__tests__/loadSupergraphSdlFromStorage.test.ts

```diff
@@ -34,253 +41,571 @@ describe('loadSupergraphSdlFromStorage', () => {
       compositionId: null,
 
     });
+    expect(result).toMatchInlineSnapshot(snapshot);
+  });
 
-    expect(result).toMatchInlineSnapshot(`
-      Object {
-        "id": "originalId-1234",
-        "supergraphSdl": "schema
-        @core(feature: \\"https://specs.apollo.dev/core/v0.2\\")
-        @core(feature: \\"https://specs.apollo.dev/join/v0.2\\", for: EXECUTION)
-        @core(feature: \\"https://specs.apollo.dev/tag/v0.1\\")
-      {
-        query: Query
-        mutation: Mutation
-      }
-
-      directive @core(feature: String!, as: String, for: core__Purpose) repeatable on SCHEMA
+  it('Queries alternate Uplink URL if first one fails', async () => {
+    mockSupergraphSdlRequest().reply(500);
+    mockSupergraphSdlRequestAlternate('originalId-1234').reply(
+      200,
+      JSON.stringify({
+        data: {
+          routerConfig: {
+            __typename: 'RouterConfigResult',
+            id: 'originalId-1234',
+            supergraphSdl: getTestingSupergraphSdl()
+          },
+        },
+      }),
+    );
 
-      directive @join__field(graph: join__Graph!, requires: join__FieldSet, provides: join__FieldSet, type: String, external: Boolean) repeatable on FIELD_DEFINITION | INPUT_FIELD_DEFINITION
+    const fetcher = getDefaultFetcher();
+    const result = await loadSupergraphSdlFromUplinks({
+      graphRef,
+      apiKey,
+      endpoints: [mockCloudConfigUrl, mockAlternateCloudConfigUrl],
+      fetcher,
+      compositionId: "originalId-1234",
+      maxRetries: 1
+    });
 
-      directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+    expect(result).toMatchInlineSnapshot(snapshot);
+    nock.cleanAll();
+  });
 
-      directive @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE
+  it('Throws error if all Uplink URLs fail', async () => {
+    mockSupergraphSdlRequest("originalId-1234").reply(500);
+    mockSupergraphSdlRequestAlternate("originalId-1234").reply(500);
 
-      directive @join__type(graph: join__Graph!, key: join__FieldSet, extension: Boolean! = false) repeatable on OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR
+    const fetcher = getDefaultFetcher();
+    await expect(
+      loadSupergraphSdlFromUplinks({
+        graphRef,
+        apiKey,
+        endpoints: [mockCloudConfigUrl, mockAlternateCloudConfigUrl],
+        fetcher,
+        compositionId: "originalId-1234",
+        maxRetries: 1
+      }),
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
+      `"An error occurred while fetching your schema from Apollo: 500 Internal Server Error"`,
+    );
+    nock.cleanAll();
```

Comment:
We shouldn't need this here if the mocks and matching requests are correct

Reply:
removed all calls to `nock.cleanAll()` in my code.